op-service/sources/eth_client.go

```diff
@@ -537,3 +537,33 @@ func (s *EthClient) PendingNonceAt(ctx context.Context, account common.Address)
 	err := s.client.CallContext(ctx, &result, "eth_getTransactionCount", account, "pending")
 	return uint64(result), err
 }
+
+// BalanceAt returns the wei balance of the given account.
+func (s *EthClient) BalanceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (*big.Int, error) {
+	var result hexutil.Big
+	var err error
+	if blockNumber != nil {
+		err = s.client.CallContext(ctx, &result, "eth_getBalance", account, blockNumber)
+	} else {
+		err = s.client.CallContext(ctx, &result, "eth_getBalance", account, "pending")
```

Comment:
What motivates a change in default behavior here? `"pending"` seems to include all valid transactions that haven't been included in a block yet but exist in the mempool (in an undefined order subject to change). This seems like it could end up leading to some unexpected behavior if the user isn't aware. If having the optionality to select a defaultBlock other than "latest" is desirable then it might be better to accept blockNumber as a string.

See: https://ethereum.org/en/developers/docs/apis/json-rpc/#default-block

The documentation is missing is that "latest" is the default.

Reply:
Fixed at https://github.com/ethereum-optimism/optimism/pull/14751/commits/8c4e101a209d7e875ec323861bc044f070d9837a